swarm/pss/pss.go

```diff
@@ -446,8 +448,25 @@ func (p *Pss) isSelfRecipient(msg *PssMsg) bool {
 
 // test match of leftmost bytes in given message to node's Kademlia address
 func (p *Pss) isSelfPossibleRecipient(msg *PssMsg) bool {
+	// in partial addressing mode self is possible recipient if To address is a prefix
+	// of the local kademlia base address
+	params := newMsgParamsFromBytes(msg.Control)
```

Comment:
Please let's not parse the struct before we check the bits.

Reply:
why? i dont get it

Reply:
Never mind, the actual content doesn't get parsed in the `newMsgParamsFromBytes` if they're not set.